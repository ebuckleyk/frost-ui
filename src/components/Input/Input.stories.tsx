import * as React from 'react';
import { Meta, StoryObj } from '@storybook/react';
import { Input } from './Input';
import { Label } from '../Label';
import { Card } from '../Card';

type ComponentType = typeof Input;
const meta: Meta<ComponentType> = {
  component: Input,
  render: () => (
    <Card className="flex items-center p-5">
      <Input />
    </Card>
  ),
};

export default meta;

type Story = StoryObj<ComponentType>;

export const Demo: Story = {
  args: {
    placeholder: 'Placeholder text',
  },
};

export const SelectWithLabel: Story = {
  render: (args) => {
    return (
      <Card className="p-10">
        <Label htmlFor="input">Input</Label>
        <Input {...args} id="input" placeholder="Placeholder text" />
      </Card>
    );
  },
};
