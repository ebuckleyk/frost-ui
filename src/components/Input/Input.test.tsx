import * as React from 'react';
import { render } from '@testing-library/react';

import { Input } from './Input';

describe('Input', () => {
  it('should render Input component to match snapshot', () => {
    const result = render(<Input />);
    expect(result).toMatchSnapshot();
  });
});
