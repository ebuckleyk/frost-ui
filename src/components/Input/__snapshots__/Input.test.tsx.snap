// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input > should render Input component to match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <input
        class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-sm shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50"
      />
    </div>
  </body>,
  "container": <div>
    <input
      class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-sm shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
